Signal/src/view controllers/PrivacySettingsTableViewController.m

```diff
@@ -174,6 +235,42 @@ - (void)didToggleScreenSecuritySwitch:(UISwitch *)sender
     BOOL enabled = self.enableScreenSecuritySwitch.isOn;
     DDLogInfo(@"%@ toggled screen security: %@", self.tag, enabled ? @"ON" : @"OFF");
     [Environment.preferences setScreenSecurity:enabled];
+    [self validateSecuritySwitches];
+}
+
+- (void)didToggleTouchIDSwitch:(UISwitch *)sender
+{
+    // Make the user verify with TouchID when enabling/disabling.
+    BOOL enabled = self.enableTouchIDSecuritySwitch.isOn;
+    DDLogInfo(@"%@ toggled touchID: %@", self.tag, enabled ? @"ON" : @"OFF");
+    __weak typeof(self) weakSelf = self;
+    [TouchIDManager.shared authenticateViaTouchIDCompletion:^(TouchIDAuthResult result) {
+        switch (result) {
+            case TouchIDAuthResultUnavailable:
+            case TouchIDAuthResultUserCanceled:
+            case TouchIDAuthResultFailed:
+                // restore switch state
+                weakSelf.enableTouchIDSecuritySwitch.on = !enabled;
+                break;
+            case TouchIDAuthResultSuccess:
+                [Environment.preferences setTouchIDEnabled:enabled];
+                if (enabled) {
+                    // If TouchID is on, Screen Security must also be on.
+                    [Environment.preferences setScreenSecurity:YES];
+                }
+                break;
+        }
+        [weakSelf validateSecuritySwitches];
+    }];
+}
+
+/// Ensures that security switches reflect the user's preferences.
+- (void)validateSecuritySwitches
```

Comment:
I don't see anything inherently wrong with this, but IMO it muddies the waters a bit about responsibilities. 

Can you elaborate on why this method exists?